File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/atts/service/schema/resolver/AttValueCtx.java
package ru.citeck.ecos.records3.record.op.atts.service.schema.resolver;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import ru.citeck.ecos.commons.data.DataValue;
import ru.citeck.ecos.commons.data.ObjectData;
import ru.citeck.ecos.records2.RecordRef;

import java.util.Map;

public interface AttValueCtx {

    @NotNull
    RecordRef getRef() throws Exception;

    @NotNull
    String getLocalId() throws Exception;

    @NotNull
    DataValue getAtt(@NotNull String attribute) throws Exception;

    @Nullable
    <T> T getAtts(@NotNull Class<T> type) throws Exception;

    @NotNull
    ObjectData getAtts(@NotNull Map<String, ?> attributes) throws Exception;
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/atts/service/schema/resolver/AttValueCtx.kt
package ru.citeck.ecos.records3.record.op.atts.service.schema.resolver

import ru.citeck.ecos.commons.data.DataValue
import ru.citeck.ecos.commons.data.ObjectData
import ru.citeck.ecos.records2.RecordRef

interface AttValueCtx {

    fun getRef(): RecordRef

    fun getLocalId(): String

    fun getAtt(attribute: String): DataValue

    fun getAtts(attributes: Map<String, *>): ObjectData

    fun <T : Any> getAtts(attributes: Class<T>): T
}
